### PR TITLE
add networkpolicy cases for kube-burner

### PR DESCRIPTION
**Pull request by mohit-sheth**
### Description
Cases are explained here https://docs.google.com/document/d/1OQtOXtqqHzV79vJ04r3z1icv5b-9y5AtWGrEgUMnvEg/edit

### CODE REVIEW

File: roles/kube-burner/files/nginx-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      labels:
+        name: nginx-{{.Iteration}}
+    spec:
+      nodeSelector:
+        {{.nodeSelectorKey}}: {{.nodeSelectorValue}}
+      containers:
+      - name: nginx
+        image: quay.io/openshift-scale/nginx
+        ports:
+        - containerPort: 80
```
Comment (rsevilla87) [inline roles/kube-burner/files/nginx-deployment.yaml:17]: Container port is 8080 for this container image.

**Comment by rsevilla87** [comment]
LGTM so far, I'm currently working on a mechanism to verify policy enforcement. Will add a commit to this PR as soon as I have something.

File: resources/operator.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
       serviceAccountName: benchmark-operator
       containers:
         - name: benchmark-operator
-          image: quay.io/benchmark-operator/benchmark-operator:master
+          image: quay.io/msheth/benchmark-operator:test
```
Comment (jtaleric) [inline resources/operator.yaml:33]: If this is ready for merge we should swap this back.

**Review by jtaleric — COMMENTED** [review]
Random placeholder about the Quay image.

**Review by jtaleric — CHANGES_REQUESTED** [review]
Can we get a test added please

**Review by sjug — APPROVED** [review]
Given the scope discussed with Mohit, this looks good to me.
We're banking entirely on the default-deny network policy application given our scraper pod is outside all created namespaces. Hopefully if the deny policy applied successfully, the rest of them have been as well. 
We don't really have any performance metrics for timing given the policies seem to be applied prior to even the first curl to the various pods.
It may be useful to have a Prometheus metric that exposes the count of currently applied network policies.

File: roles/kube-burner/files/case2-networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: case2-{{.podselector_label}}{{.Replica}}-{{.ingress_label}}-{{randInteger 0 1000000}}
```
Comment (rsevilla87) [inline roles/kube-burner/files/case2-networkpolicy.yaml:4]: {{randInteger 0 1000000}} is not necessary in all networkpolicy objects

**Review by rsevilla87 — COMMENTED** [review]
Added some comments

File: roles/kube-burner/files/cross-ns-networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: cross-ns-{{randInteger 1 1000000}}
+spec:
+  podSelector:
+    matchLabels:
+      label1: foo-{{randInteger 1 5}}
```
[the remark this answers is not in the document]
Comment (mohit-sheth) [reply roles/kube-burner/files/cross-ns-networkpolicy.yaml:8]: yes, I am trying to follow
```
For each unique label in each Namespace, create a NetworkPolicy with that label as the podSelector, allowing traffic from pods with some other randomly-selected Pod label and randomly-selected Namespace label.
```

hence using the randinteger

**Pull request merged**